docs/specifications/1-introduction.md

```diff
@@ -0,0 +1,172 @@
+
+* *jsii* APIs strive to feel idiomatic in all supported languages.
+* *jsii* applications behave identically regardless of the language they are
+  written in. It favors correctness over performance.
+* *jsii* **does not** attempt to support all TypeScript idioms (many features of
+  TypeScript cannot be expressed in some target languages).
+  * When prohibiting an idiom, *jsii* strives to provide an error message that
+    gives the user insight into why the pattern cannot be supported.
+* *jsii* does not force API design opinions on the developer:
+  * Reserved names are limited to a minimum.
+  * TypeScript API design patterns that will not transpose well in other
+    languages may cause a warning to be issued, but the developer is ultimately
+    entitled to decide whether they want to take or leave the advice.
+* *jsii* uses idiomatic tools whenever possible:
+  * Generated libraries can be easily published to the "standard" package
+    repository for the language.
+  * Standard tools can be used to work with the generated libraries, and do not
+    require any special configuration.
+
+
+## Concepts
+
+*jsii* allows developers to author code once in **TypeScript**, while allowing
+use in a variety of other programming languages (including **C#**, **Java** and
+**Python**).
+
+### Assemblies
+
+The *jsii Assembly* document contains a specific representation of the API
+exported by the **TypeScript** module. Similar to a header file in the **C++**
+world, it contains only information about the signatures of APIs (type names
+with method and property signatures, ...) and no implementation.
+
+The *npm* package produced as a result of compiling the **TypeScript** source
+remains the source of truth with respects to implementation of the API.
+
+### Host & Kernel
+
+The [*jsii* runtime architecture] defines two processes:
+
+1. The *host* process runs the users' code natural environment (a Java virtual
+   machine, the .NET Runtime, ...).
+2. The *kernel* process hosts the **JavaScript** code from the standard *npm*
+   package.
+
+The *host* process is responsible for starting the *kernel* process as needed,
+and a designated *host runtime library* is provided to make this transparent to
+the user.
+
+The two processes exchange messages over a designated communication channel (for
+exmaple, using pipes), using a *kernel API* that is standardized in the *jsii
+specification*.
+
+[*jsii* runtime architecture]: ../runtime-architecture.md
+
+
+## Components
+
+Several tools are involved in making this possible:
+
+* [`jsii`] is a modified **TypeScript** compiler. In addition to generating
+  **JavaScript** code from the source, it produces a *jsii Assembly* document.
+* [`jsii-pacmak`] generates language bindings from a package compiled using
+  `jsii`. It generates code in *host* languages that expose the API declared in
+  the *jsii Assembly* document.
+* *Host runtime libraries* provide basic features that the code generated by
```

Comment:
I'm a bit lost on this host runtime library and why it's needed; all the others are fairly clear to me. Can you be clearer?